frontends/web/src/routes/account/walletconnect/connect.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FormEvent, useCallback, useEffect, useState } from 'react';
+import { useLoad } from '../../../hooks/api';
+import * as accountApi from '../../../api/account';
+import { Header, Main } from '../../../components/layout';
+import { SignClientTypes } from '@walletconnect/types';
+import useInitialization, { pair, web3wallet } from './utils';
+import { alertUser } from '../../../components/alert/Alert';
+import { View, ViewContent } from '../../../components/view/view';
+import { WCHeader } from './components/header/header';
+import { WCConnectForm } from './components/connect-form/connect-form';
+import { TConnectStatus } from './types';
+import { WCIncomingPairing } from './components/incoming-pairing/incoming-pairing';
+import { WCSuccessPairing } from './components/success-pairing/success-pairing';
+import styles from './connect.module.css';
+
+type TProps = {
+  code: string;
+  accounts: accountApi.IAccount[]
+};
+
+export const ConnectScreenWalletConnect = ({
+  code,
+  accounts
+}: TProps) => {
+  const initialized = useInitialization();
+  const [uri, setUri] = useState('');
+  const [status, setStatus] = useState<TConnectStatus>('connect');
+  const [currentProposal, setCurrentProposal] = useState<SignClientTypes.EventArguments['session_proposal']>();
+  const receiveAddresses = useLoad(accountApi.getReceiveAddressList(code));
+  const onSessionProposal = useCallback(
+    (proposal: SignClientTypes.EventArguments['session_proposal']) => {
+      setStatus('incoming_pairing');
+      setCurrentProposal(proposal);
+    },
+    []
+  );
+
+  useEffect(() => {
+    web3wallet?.on('session_proposal', onSessionProposal);
+    return () => {
+      web3wallet?.off('session_proposal', onSessionProposal);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [onSessionProposal, web3wallet]);
```

Comment:
it doesn't complain to me if it is just `}, [onSessionProposal]);` without `// eslint-disable-next-line react-hooks/exhaustive-deps`, or do you think it is better to have web3wallet in the list?

Reply:
Hm yeah. I'll put initialized instead and only starts listening when initialized === true, and use initialized as dep.

```js
  const initialized = useInitialization(); // initialising web3

useEffect...
   if (initialized) {
      web3wallet?.on('session_proposal', onSessionProposal);
      return () => {
        web3wallet?.off('session_proposal', onSessionProposal);
      };
    }

```
Previously we needed to put web3wallet as dep because it's always undefined at first. So if we don't put it as dep, the whole thing is not going to work (as it stays undefined, so the .on() wont even be there). But if we check for initialized, it should be fine as the hook creates **initialises** the web3 wallet. I'll look into if it's possible to refactor the hook ,soon :)